server/room.ts

```diff
@@ -266,45 +186,51 @@ export class Room {
   getHostState() {
     return {
       video: this.video,
+      videoDuration: this.videoDuration,
       videoTS: this.videoTS,
       paused: this.paused,
     };
   }
 
-  async resetRoomVM() {
-    const id = this.vBrowser && this.vBrowser.id;
-    this.vBrowser = undefined;
-    this.roster.forEach((user, i) => {
-      this.roster[i].isController = false;
-    });
-    this.cmdHost(undefined, '');
-    if (id) {
-      try {
-        await resetVM(id);
-      } catch (e) {
-        console.error(e);
-      }
+  cmdHost(
+    socket: Socket,
+    data?: PlaylistVideo,
+    options = { skipMessage: false }
+  ) {
+    if (!socket) {
+      return;
+    }
+
+    if (data) {
+      // remove next hosted video from the playlist
+      this.videoPlaylist = this.videoPlaylist.filter(
+        (video) => video.url !== data.url
+      );
+      socket.emit('playlistUpdate', this.videoPlaylist);
     }
-  }
 
-  cmdHost(socket: Socket | undefined, data: string) {
-    this.video = data;
+    this.video = data ? data.url : '';
     this.videoTS = 0;
+    this.videoDuration = data ? data.duration : 0;
     this.paused = false;
     this.tsMap = {};
     this.io.of(this.roomId).emit('REC:tsMap', this.tsMap);
     this.io.of(this.roomId).emit('REC:host', this.getHostState());
-    if (socket && data) {
+    if (socket && data && !options.skipMessage) {
       const chatMsg = { id: socket.id, cmd: 'host', msg: data };
       this.addChatMessage(socket, chatMsg);
     }
   }
 
-  addChatMessage(socket: Socket | undefined, chatMsg: any) {
+  addChatMessage(socket: Socket, chatMsg: any) {
```

Comment:
I believe at some point there was a caller that didn't pass socket but we still want to update the chat. May want to double-check that before applying this change.

Reply:
Nope, from what I can see all addChatMessage calls now pass a socket as the first arg